libcontainer/intelrdt/intelrdt_test.go

```diff
@@ -98,31 +100,187 @@ func TestIntelRdtSetMemBwScSchema(t *testing.T) {
 }
 
 func TestApply(t *testing.T) {
-	helper := NewIntelRdtTestUtil(t)
-
 	const closID = "test-clos"
-	closPath := filepath.Join(helper.IntelRdtPath, closID)
 
-	helper.config.IntelRdt.ClosID = closID
-	intelrdt := newManager(helper.config, "container-1", closPath)
-	if err := intelrdt.Apply(1234); err == nil {
-		t.Fatal("unexpected success when applying pid")
-	}
-	if _, err := os.Stat(closPath); err == nil {
-		t.Fatal("closid dir should not exist")
+	tests := []struct {
+		name     string
+		testFunc func(*intelRdtTestUtil)
+	}{
+		{
+			name: "failure because non-pre-existing CLOS",
+			testFunc: func(helper *intelRdtTestUtil) {
+				helper.config.IntelRdt = &configs.IntelRdt{
+					ClosID: closID,
+				}
+
+				closPath := filepath.Join(intelRdtRoot, closID)
+				intelrdt := newManager(helper.config, "", closPath)
+				if err := intelrdt.Apply(1234); err == nil {
+					t.Fatal("unexpected success when applying pid")
+				}
```

Comment:
It seems this can be part of the `for ... range tests` loop,a s it's c&p in every  `testFunc()`. Yes, one needs a slightly different param, but that can be part of the test struct, I think.